app/src/main/java/com/babylon/wallet/android/presentation/dialogs/info/InfoDialog.kt

```diff
@@ -0,0 +1,270 @@
+import org.intellij.markdown.MarkdownElementTypes.INLINE_LINK
+import org.intellij.markdown.MarkdownElementTypes.LINK_DESTINATION
+import org.intellij.markdown.MarkdownElementTypes.LINK_LABEL
+import org.intellij.markdown.MarkdownElementTypes.LINK_TEXT
+import org.intellij.markdown.MarkdownTokenTypes
+import org.intellij.markdown.ast.ASTNode
+import org.intellij.markdown.ast.findChildOfType
+import org.intellij.markdown.ast.getTextInNode
+
+typealias DSR = com.babylon.wallet.android.designsystem.R.drawable
+
+@Composable
+fun InfoDialog(
+    modifier: Modifier = Modifier,
+    viewModel: InfoViewModel,
+    onDismiss: () -> Unit
+) {
+    val glossaryItem by viewModel.glossaryItem.collectAsStateWithLifecycle()
+
+    BottomSheetDialogWrapper(
+        modifier = modifier,
+        showDragHandle = true,
+        dragToDismissEnabled = true,
+        onDismiss = onDismiss
+    ) {
+        InfoDialogContent(
+            scrollState = rememberScrollState(),
+            markdownContent = glossaryItem?.resolveTextFromGlossaryItem(),
+            drawableRes = glossaryItem?.resolveIconFromGlossaryItem(),
+            onGlossaryItemClick = viewModel::onGlossaryItemClick,
+        )
+    }
+}
+
+@Composable
+private fun InfoDialogContent(
+    modifier: Modifier = Modifier,
+    scrollState: ScrollState,
+    markdownContent: String?,
+    @DrawableRes drawableRes: Int?,
+    onGlossaryItemClick: (String) -> Unit,
+) {
+    val coroutineScope = rememberCoroutineScope()
+    val context = LocalContext.current
+
+    val customHeading2: MarkdownComponent = { markdownComponentModel ->
+        markdownComponentModel.node.findChildOfType(MarkdownTokenTypes.ATX_CONTENT)?.let { markDownNode ->
+            val styledText = buildAnnotatedString {
+                pushStyle(RadixTheme.typography.title.toSpanStyle().copy(color = RadixTheme.colors.gray1))
+                buildMarkdownAnnotatedString(markdownComponentModel.content, markDownNode)
+                pop()
+            }
+            Text(
+                styledText,
+                modifier = Modifier.fillMaxSize(),
+                textAlign = TextAlign.Center
+            )
+        }
+    }
+
+    val linkAnnotator = markdownAnnotator { content, child ->
+        if (child.type == INLINE_LINK) {
+            val linkText = child.findChildOfType(LINK_TEXT)?.children?.innerList()?.firstOrNull()
+            val destination = child.findChildOfType(LINK_DESTINATION)
+                ?.getTextInNode(content)
+                ?.toString()
+            val linkLabel = child.findChildOfType(LINK_LABEL)
+                ?.getTextInNode(content)?.toString()
+            val annotation = destination ?: linkLabel
+            if (annotation != null) pushStringAnnotation(MARKDOWN_TAG_URL, annotation)
+
+            pushStyle(
+                SpanStyle(
+                    fontWeight = FontWeight.Bold,
+                    color = Color(0xFF052CC0) // RadixTheme.colors.blue2
```

Comment:
Instead of hardcoding, we could import `Blue2` const from `RadixColors.kt`

Reply:
oh I just wanted to explain to which theme color the color code corresponds, because the annotator there is not a composable therefore I can't access the theme.

Reply:
That's why we can use the const directly instead of the composable value from the theme.

Reply:
ohh I am so silly sometimes.... yes you are right